src/librbd/AioImageRequest.cc

```diff
@@ -724,10 +732,140 @@ void AioImageFlush<I>::send_image_cache_request() {
   image_ctx.image_cache->aio_flush(req_comp);
 }
 
+template <typename I>
+void AioImageWriteSame<I>::assemble_extent(const ObjectExtent &object_extent,
```

Comment:
Yes, I should really take striping into consideration. I didn't do that just to make things work as soon as possible. I will try to handle the stripping stuff in next round.